pkg/platform/kube/controller/cronjobmonitoring.go

```diff
@@ -34,41 +34,41 @@ func NewCronJobMonitoring(parentLogger logger.Logger,
 	return newCronJobMonitoring
 }
 
-func (cjpd *CronJobMonitoring) start() {
+func (cjm *CronJobMonitoring) start() {
 
-	go cjpd.startStaleCronJobPodsDeletionLoop() // nolint: errcheck
+	go cjm.startStaleCronJobPodsDeletionLoop() // nolint: errcheck
 
 }
 
 // delete all stale cron job pods (as k8s lacks this logic, and CronJob pods are never deleted)
-func (cjpd *CronJobMonitoring) startStaleCronJobPodsDeletionLoop() error {
-	stalePodsFieldSelector := cjpd.compileStalePodsFieldSelector()
+func (cjm *CronJobMonitoring) startStaleCronJobPodsDeletionLoop() error {
+	stalePodsFieldSelector := cjm.compileStalePodsFieldSelector()
 
-	cjpd.logger.InfoWith("Starting stale cron job pods deletion loop",
-		"staleCronJobPodsDeletionInterval", cjpd.staleCronJobPodsDeletionInterval,
+	cjm.logger.InfoWith("Starting stale cron job pods deletion loop",
+		"staleCronJobPodsDeletionInterval", cjm.staleCronJobPodsDeletionInterval,
 		"fieldSelectors", stalePodsFieldSelector)
 
 	for {
 
 		// sleep until next deletion time staleCronJobPodsDeletionInterval
-		time.Sleep(*cjpd.staleCronJobPodsDeletionInterval)
+		time.Sleep(*cjm.staleCronJobPodsDeletionInterval)
 
-		err := cjpd.controller.kubeClientSet.
+		err := cjm.controller.kubeClientSet.
 			CoreV1().
-			Pods(cjpd.controller.namespace).
+			Pods(cjm.controller.podNamespace).
```

Comment:
Setting it as `podNamespace` means we'll never clean cron-job pods of functions that run outside of the controller's pod namespace - which could happen if the `CRDNamespace` is '*' (or any other namespace that's != podNamespace`) - because on function creation we can pass any namespace

Reply:
@sahare92  - we have talked about this and you claimed the ctonjobs are meant to be created in the controller's namespace, NOT the function's namespace.

Reply:
They're created in the function namespace today: https://github.com/nuclio/nuclio/blob/6d75da4bd73a582f2543730dea23dc35fd03e26c/pkg/platform/kube/functionres/lazy.go#L1190

Reply:
We'll have to decide which way to resolve this